subprojects/docs/src/docs/userguide/upgrading_version_4.adoc

```diff
@@ -328,6 +377,149 @@ To override this behavior, you can explicitly declare the configuration to use i
 
 == Changes in detail
 
+[[rel5.0:default_memory_settings]]
+=== [5.0] Default memory settings changed
+
+The command line client now starts with 64m of heap instead of 1g.
+This may affect builds running directly inside the client VM using `--no-daemon` mode.
+We discourage the use of `--no-daemon`, but if you must use it, you can increase the available memory using the `GRADLE_OPTS` environment variable.
+
+The Gradle daemon now starts with 512m of heap instead of 1g.
+Large projects may have to increase this setting using the `org.gradle.jvmargs` property.
+
+All workers, including compilers and test executors, now start with 512m of heap. The previous default was 1/4th of physical memory.
+Large projects may have to increase this setting on the relevant tasks, e.g. `JavaCompile` or `Test`.
+
+[[rel5.0:default_tool_versions]]
+=== [5.0] New default versions for code quality plugins
+
+The default tool versions of the following code quality plugins have been updated:
+
+ * The http://checkstyle.sourceforge.net[Checkstyle] plugin now uses http://checkstyle.sourceforge.net/releasenotes.html#Release_8.12[8.12] instead of 6.19 by default.
```

Comment:
This was something I was debating and ended up with this syntax. I'm linking to the technologies not the Gradle plugins. Is it confusing if I include the word plugin in the link?

Reply:
It's a tricky one. If I wanted to focus on the tool itself, I would use "The plugin for Checkstyle ...", retaining the link on "Checkstyle". Having just the link on "Checkstyle" in the name "Checkstyle Plugin" would look weird.

Keeping "plugin" lowercase runs into the issues that dissuade me from using it, even though the uppercase "Plugin" looks high ceremony and a bit over the top. "plugin" could refer to something that plugs into Checkstyle rather than Gradle. And we don't talk about "the plugin called Checkstyle" because its name is Checkstyle Plugin to distinguish it from the tool it integrates with. I'm not a huge fan of the naming convention, but I couldn't really think of anything better that works in all cases without ambiguity.

Reply:
Hope that helps :smile: